src/engine/Match_search_mode.ml

```diff
@@ -577,7 +577,14 @@ let rec filter_ranges (env : env) (xs : (RM.t * MV.bindings list) list)
          | R.CondType (mvar, opt_lang, _, ts) -> (
              let mvalue_to_expr m =
                match Metavariable.mvalue_to_any m with
+               (* Capturing AST_generic.Id with the metavariable
```

Comment:
I think it's better to replace this comment with what you told me during our 1:1. What you say here one can understand by looking at `mvalue_to_any`. Where did you say that Pro generated the `IdQualified` ?

Reply:
Although I'm not sure you were right about this. I managed to reproduce the issue with OSS using this rule:

```yaml
    patterns:
      - patterns:
          - pattern: |
              @$X
          - metavariable-type:
              metavariable: $X
              type: foo
```

and this code

```python
@x.bar()
def test():
    pass
```

Reply:
Thanks, that test case is useful. Added to the PR.

I also updated the comment accordingly. I didn't mention the Pro engine since this bug is not specific to the Pro engine now.